codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -277,6 +278,23 @@ public boolean visit(Http2Stream stream) {
         }
     }
 
+    @Override
+    public void headersSent(int streamId, boolean sent) {
+        Http2Stream stream = stream(streamId);
+        assert stream != null : "Invalid stream id";
+        stream.setProperty(headersSentKey, sent);
+    }
+
+    @Override
+    public boolean headersSent(int streamId) {
+        Http2Stream stream = stream(streamId);
+        if (stream == null) {
+            return false;
```

Comment:
Or rather, if we cannot get a stream and cannot create one (becuase streamId < nextStreamIdToCreate) . Updated the comment in the code to explain this.